### PR TITLE
Boot time loop detection

### PR DESCRIPTION
**Pull request by dhellmann**

### CODE REVIEW

**Comment by dhellmann** [comment]
This includes some of Ryan's changes from another pull request, and it needs more tests, but I wanted to put it up for folks to see instead of waiting.

File: akanda/rug/main.py [hunk 1]
```diff
@@ -205,6 +206,21 @@ def main(argv=sys.argv[1:]):
     # bring the mgt tap interface up
     quantum.ensure_local_service_port()
 
+    # Ask keystone for the UUID for our admin user so we can filter
+    # events we create ourselves.
```
Comment (ryanpetrello) [inline akanda/rug/main.py]: s/we create ourselves/created by akanda itself

File: akanda/rug/notifications.py [hunk 2]
```diff
@@ -130,12 +131,19 @@ def _make_event_from_message(message):
         else:
             # LOG.debug('ignoring message %r', message)
             return None
+
+    if context_user and ignore_user and crud != event.COMMAND:
+        if context_user == ignore_user:
+            # Ignore events generated by operations taken by this service.
+            # LOG.debug('Ignoring %s event from our admin user', event_type)
```
Comment (ryanpetrello) [inline akanda/rug/notifications.py]: Edit/uncomment this?

File: akanda/rug/vm_manager.py [hunk 3]
```diff
@@ -112,13 +112,18 @@ def update_state(self, worker_context, silent=False):
...
+            # update_state() is called from Alive() to check the
+            # status of the router. If we can't talk to the API at
+            # that point, the router should be considered missing and
+            # we should reboot it, so mark it down if we think it was
+            # configured before.
+            if old_state == CONFIGURED:
+                self.log.debug(
+                    'Did not find router alive, marking it as down',
+                )
+                self.state = DOWN
```
Comment (ryanpetrello) [inline akanda/rug/vm_manager.py:126]: Will this overwrite a state of `ERROR`?  Do we need `if old_state == CONFIGURED and self.state != ERROR:` ?

File: akanda/rug/vm_manager.py [hunk 4]
```diff
@@ -300,6 +305,7 @@ def configure(self, worker_context, failure_state=RESTART, attempts=None):
                     cfg.CONF.akanda_mgt_service_port,
                     config
                 )
+                # raise RuntimeError('testing with fake failure')
```
Comment (ryanpetrello) [inline akanda/rug/vm_manager.py]: Remove?

**Pull request merged**